src/Core/Content/Media/Core/Application/RemoteThumbnailLoader.php

```diff
@@ -168,19 +171,28 @@ private function getBaseUrl(): string
         return \rtrim($this->filesystem->publicUrl(''), '/');
     }
 
-    private function getUrl(string $mediaUrl, string $mediaPath, string $width, string $height, ?\DateTimeInterface $mediaUpdatedAt): string
+    private function getUrl(Entity $mediaEntity, string $mediaUrl, string $width, string $height): ?string
```

Comment:
I had a look again, and the type hint should be the following:

```suggestion
    private function getUrl(MediaEntity|PartialEntity $mediaEntity, string $mediaUrl, string $width, string $height): ?string
```

I know you had that before, and PHPStan is complaining about it, but we would just need to fix the other places. If you look here, it could only be those two possibilities https://github.com/shopware/shopware/blob/trunk/src/Core/Content/DependencyInjection/media_path.xml#L13-L14

Reply:
If you do not feel confident with this change, I can push a commit, if you like